File: 1.1/1.1.pl
member(X, [X|_]).
member(X, [_|T]) :- member(X, T).

not_member(_, []).
not_member(X, [H|T]) :- X \= H, not_member(X, T).

count_occurrences(_, [], 0).
count_occurrences(X, [X|T], N) :- count_occurrences(X, T, N1), N is N1 + 1.
count_occurrences(X, [_|T], N) :- count_occurrences(X, T, N).


filter_duplicates([], []).
filter_duplicates([H|T], Result) :-
    count_occurrences(H, [H|T], Count),
    (Count =:= 2 -> Result = [H|Rest]; Result = Rest),
    filter_duplicates(T, Rest).

remove_elements([], _, []).
remove_elements([H|T], ToKeep, Result) :-
    not_member(H, ToKeep),
    !,
    remove_elements(T, ToKeep, Result).
remove_elements([H|T], ToKeep, [H|Result]) :-
    remove_elements(T, ToKeep, Result).


main :-
    process,
    halt.

process :-
    write('Введіть список цілих чисел (завершiть введення "." на новому рядку): '),
    read(List),
    filter_duplicates(List, ToKeep),
    remove_elements(List, ToKeep, Result),
    write('Результат: '), write(Result), nl.

:- main.

File: 1.1/1.1.hs
import Data.List

filterDuplicates :: Eq a => [a] -> [a]
filterDuplicates xs = [x | x <- nub xs, length (elemIndices x xs) == 2]

removeNotInList :: Eq a => [a] -> [a] -> [a]
removeNotInList list1 list2 = list1 `intersect` list2

main :: IO ()
main = do
  putStrLn "Введіть список цілих чисел через пpобіл:"
  input <- getLine
  let numbers = map read (words input) :: [Int]
      toKeep = filterDuplicates numbers
      result = removeNotInList numbers toKeep
  putStrLn $ "Результат: " ++ unwords (map show result)